File: src/main/java/yunas/handler/AppServlet.java
package yunas.handler;

import yunas.exception.YunasExceptionProvider;
import yunas.http.HttpMethod;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

import static yunas.handler.Execute.execute;

/**
 * Handle All Requests.
 *
 * @author Yosuke Kobayashi
 */
public class AppServlet extends HttpServlet {


    @Override
    public void doGet(HttpServletRequest request,
                      HttpServletResponse response) throws ServletException, IOException{

        execute(request, response, HttpMethod.GET);

    }

    @Override
    protected void doHead(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {

        execute(req, resp, HttpMethod.HEAD);
    }

    @Override
    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {

        execute(req, resp, HttpMethod.POST);
    }

    @Override
    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {

        execute(req, resp, HttpMethod.PUT);
    }

    @Override
    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {

        execute(req, resp, HttpMethod.DELETE);
    }

    @Override
    protected void doOptions(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {

        execute(req, resp, HttpMethod.OPTIONS);
    }

    @Override
    protected void doTrace(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        // No Support
        throw new YunasExceptionProvider().notFoundPath(req.getPathInfo());
    }

}

File: src/main/kotlin/yunas/handler/AppServlet.kt
package yunas.handler

import yunas.exception.YunasExceptionProvider
import yunas.handler.Execute.execute
import yunas.http.HttpMethod
import java.io.IOException
import javax.servlet.ServletException
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * Handle All Requests.
 *
 * @author Yosuke Kobayashi
 */
class AppServlet : HttpServlet() {


    @Throws(ServletException::class, IOException::class)
    public override fun doGet(request: HttpServletRequest,
                              response: HttpServletResponse) {

        execute(request, response, HttpMethod.GET)

    }

    @Throws(ServletException::class, IOException::class)
    override fun doHead(req: HttpServletRequest, resp: HttpServletResponse) {

        execute(req, resp, HttpMethod.HEAD)
    }

    @Throws(ServletException::class, IOException::class)
    override fun doPost(req: HttpServletRequest, resp: HttpServletResponse) {

        execute(req, resp, HttpMethod.POST)
    }

    @Throws(ServletException::class, IOException::class)
    override fun doPut(req: HttpServletRequest, resp: HttpServletResponse) {

        execute(req, resp, HttpMethod.PUT)
    }

    @Throws(ServletException::class, IOException::class)
    override fun doDelete(req: HttpServletRequest, resp: HttpServletResponse) {

        execute(req, resp, HttpMethod.DELETE)
    }

    @Throws(ServletException::class, IOException::class)
    override fun doOptions(req: HttpServletRequest, resp: HttpServletResponse) {

        execute(req, resp, HttpMethod.OPTIONS)
    }

    @Throws(ServletException::class, IOException::class)
    override fun doTrace(req: HttpServletRequest, resp: HttpServletResponse) {
        // No Support
        throw YunasExceptionProvider().notFoundPath(req.pathInfo)
    }

}